x/multi-staking/keeper/invartiants_test.go

```diff
@@ -0,0 +1,161 @@
+				valPubKey2 := priv.PubKey()
+				bondAmount := sdk.NewCoin(MultiStakingDenomA, sdk.NewInt(500))
+				createMsg2 := stakingtypes.MsgCreateValidator{
+					Description: stakingtypes.Description{
+						Moniker:         "test",
+						Identity:        "test",
+						Website:         "test",
+						SecurityContact: "test",
+						Details:         "test",
+					},
+					Commission: stakingtypes.CommissionRates{
+						Rate:          sdk.MustNewDecFromStr("0.05"),
+						MaxRate:       sdk.MustNewDecFromStr("0.1"),
+						MaxChangeRate: sdk.MustNewDecFromStr("0.1"),
+					},
+					MinSelfDelegation: sdk.NewInt(200),
+					DelegatorAddress:  delAddr.String(),
+					ValidatorAddress:  valAddr2.String(),
+					Pubkey:            codectypes.UnsafePackAny(valPubKey2),
+					Value:             bondAmount,
+				}
+
+				_, err := suite.msgServer.CreateValidator(suite.ctx, &createMsg2)
+				suite.Require().NoError(err)
+
+				multiStakingMsg := stakingtypes.NewMsgBeginRedelegate(delAddr, valAddr, valAddr2, bondAmount)
+				_, err = suite.msgServer.BeginRedelegate(suite.ctx, multiStakingMsg)
+				suite.Require().NoError(err)
+			},
+			expPass: true,
+		},
+		{
+			name: "Success Undelegate",
+			malleate: func() {
+				bondAmount := sdk.NewCoin(MultiStakingDenomA, sdk.NewInt(250))
+				multiStakingMsg := stakingtypes.NewMsgUndelegate(delAddr, valAddr, bondAmount)
+				_, err := suite.msgServer.Undelegate(suite.ctx, multiStakingMsg)
+				suite.Require().NoError(err)
+
+				bondAmount1 := sdk.NewCoin(MultiStakingDenomA, sdk.NewInt(500))
+				multiStakingMsg1 := stakingtypes.NewMsgUndelegate(delAddr, valAddr, bondAmount1)
+				_, err = suite.msgServer.Undelegate(suite.ctx, multiStakingMsg1)
+				suite.Require().NoError(err)
+			},
+			expPass: true,
+		},
+	}
+	for _, tc := range testCases {
+		suite.SetupTest() // reset
+
+		valCoins := sdk.NewCoins(sdk.NewCoin(MultiStakingDenomA, sdk.NewInt(10000)), sdk.NewCoin(MultiStakingDenomB, sdk.NewInt(10000)))
+		err := suite.FundAccount(delAddr, valCoins)
+		suite.Require().NoError(err)
+
+		suite.msKeeper.SetBondWeight(suite.ctx, MultiStakingDenomA, sdk.MustNewDecFromStr("0.3"))
+		bondAmount := sdk.NewCoin(MultiStakingDenomA, sdk.NewInt(3001))
+		msg := stakingtypes.MsgCreateValidator{
+			Description: stakingtypes.Description{
+				Moniker:         "test",
+				Identity:        "test",
+				Website:         "test",
+				SecurityContact: "test",
+				Details:         "test",
+			},
+			Commission: stakingtypes.CommissionRates{
+				Rate:          sdk.MustNewDecFromStr("0.05"),
+				MaxRate:       sdk.MustNewDecFromStr("0.1"),
+				MaxChangeRate: sdk.MustNewDecFromStr("0.05"),
+			},
+			MinSelfDelegation: sdk.NewInt(1),
+			DelegatorAddress:  delAddr.String(),
+			ValidatorAddress:  valAddr.String(),
+			Pubkey:            codectypes.UnsafePackAny(valPubKey),
+			Value:             bondAmount,
+		}
+
+		_, err = suite.msgServer.CreateValidator(suite.ctx, &msg)
+		suite.Require().NoError(err)
+
+		tc.malleate()
```

Comment:
why there's no fail case in this test

Reply:
invariants should be success in any case

Reply:
there should be fail case in test, just set data in the store that causes it to fail